src/Akeneo/Connectivity/Connection/back/Application/ErrorManagement/Service/CollectApiError.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Application\ErrorManagement\Service;
+
+use Akeneo\Connectivity\Connection\Application\ConnectionContextInterface;
+use Akeneo\Connectivity\Connection\Application\ErrorManagement\Command\UpdateConnectionErrorCountCommand;
+use Akeneo\Connectivity\Connection\Application\ErrorManagement\Command\UpdateConnectionErrorCountHandler;
+use Akeneo\Connectivity\Connection\Domain\ErrorManagement\Model\Write\ApiErrorInterface;
+use Akeneo\Connectivity\Connection\Domain\ErrorManagement\Model\Write\HourlyErrorCount;
+use Akeneo\Connectivity\Connection\Domain\ValueObject\HourlyInterval;
+use Akeneo\Connectivity\Connection\Domain\ErrorManagement\ErrorTypes;
+use Akeneo\Connectivity\Connection\Domain\ErrorManagement\Persistence\Repository\BusinessErrorRepository;
+use Akeneo\Connectivity\Connection\Domain\Settings\Model\ValueObject\FlowType;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+/**
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class CollectApiError
+{
+    /** @var BusinessErrorRepository */
+    private $repository;
+
+    /** @var ConnectionContextInterface */
+    private $connectionContext;
+
+    /** @var ExtractErrorsFromHttpExceptionInterface */
+    private $extractErrorsFromHttpException;
+
+    /** @var UpdateConnectionErrorCountHandler */
+    private $updateErrorCountHandler;
+
+    /** @var array */
```

Comment:
```suggestion
    /** @var array = [ $errorType => ApiErrorInterface[] ] */
```

Or you could create a simple ApiErrorCollection to do everything:
- Adding error (indexed by type internally)
- Get the count of all the errors
- Get all the errors of a specific type
It will simplify this service by moving elsewhere all the glue code to handle the list of error

```suggestion
    /** @var ApiErrorCollection */
```